test/system/SmokeTestGame.cpp

```diff
@@ -158,9 +165,45 @@ BOOST_AUTO_TEST_CASE(host_server) {
                 BOOST_TEST_MESSAGE("Industry: " << planet->GetMeter(MeterType::METER_INDUSTRY)->Current());
                 BOOST_REQUIRE_LT(0.0, planet->GetMeter(MeterType::METER_RESEARCH)->Current());
                 BOOST_TEST_MESSAGE("Research: " << planet->GetMeter(MeterType::METER_RESEARCH)->Current());
-                found_planet = true;
+                home_planet = planet;
+            }
+            BOOST_REQUIRE(home_planet != nullptr);
+
+            // enqueue Troop Ship
+            ScriptingContext context;
+
+            static constexpr boost::uuids::uuid troop_ship_uuid{0x08, 0xa5, 0x8b, 0x08, 0x09, 0x29, 0x49, 0x6d, 0x84, 0xfc, 0xfa, 0xa9, 0x14, 0x24, 0xca, 0x02};
+            for (const auto* design : GetPredefinedShipDesignManager().GetOrderedShipDesigns()) {
+                BOOST_TEST_MESSAGE("Predefined ship design " << design->Name() << " " << design->UUID());
+                if (design->UUID() == troop_ship_uuid) {
+                    BOOST_REQUIRE_EQUAL(troop_design_id, INVALID_DESIGN_ID);
+                    troop_design_id = my_empire->AddShipDesign(*design, GetUniverse());
```

Comment:
prefer `context` getters to free functions like `GetUniverse`

Reply:
Fixed.
Do `GetPredefinedShipDesignManager` free function have non-free version?